File: 프로그래머스/lv2/12980. 점프와 순간 이동/점프와 순간 이동.py
def solution(n):
	if n==1:
		return 1
	else:
		answer=0
		while n!=0:
			# 순간이동이 가능할 때
			if n%2==0:
				#print("순간이동")
				n=n/2
			# 순간이동이 불가능할 때
			else:
				n-=1
				answer+=1
		return answer

File: 프로그래머스/2/12980. 점프와 순간 이동/점프와 순간 이동.java
import java.util.*;

public class Solution {
    public int solution(int n) {
        if(n==1){
            return 1;
        }else{
         int answer = 0;
            while(n!=0){
                if(n%2==0){
                    n=n/2;
                }else{
                    n--;
                    answer+=1;
                }
            }
            return answer;
        }
    }
}